feat(core): implementing collecting requestSchema and convert to json

oneweeeek opened the pull request:
<!--  
Thank you for contributing to the API GHOST project.  
This document provides guidelines to ensure smooth collaboration and maintain high code quality.  
-->

<!--
PR Title Format Guideline

Use the following format for your PR title:

    type(scope): concise description

Examples:
    feat(api): add scenario execution support
    fix(parser): resolve YAML parsing error
    docs(readme): update CLI usage instructions

Available types:
    feat:     Add a new feature
    fix:      Fix a bug
    build:    Modify build system or external dependencies
    chore:    Modify configuration files unrelated to source or test files (e.g., .gitignore, .editorconfig)
    ci:       Update CI configuration files and scripts
    test:     Add or update tests
    docs:     Update documentation (e.g., README)
    refactor: Code changes that neither fix a bug nor add a feature
    style:    modify CSS styles
-->

### Related Issue

- #5 

### Description

- Analyze Spring controller methods to collect DTOs annotated with @RequestBody and @ModelAttribute
- Implement recursive JSON conversion logic using Jackson and ClassGraph to analyze DTO structures
- Add support for nested objects and collection types
- Prevent infinite recursion by tracking visited classes with a visited set


### Testing
![image](https://github.com/user-attachments/assets/fff466ad-f990-4b97-aeec-a590c9323bd0)
![image](https://github.com/user-attachments/assets/17d30b52-df8e-47b5-aa8f-23f98bf3e5cb)

- Manually verify that JSON is generated normally even with cyclic references or complex types
- Further test cases will be added in the future
### Additional Notes

### Pre-Submission Checklist

- [x] Have you completed code style (convention) checks locally?
- [x] Have you passed the CI tests in your local environment?

oneweeeek committed:
feat: api-collector
collecting requestSchema and converting to json

sung-jun98 commented:
You’ve done an incredible job. seriously, hats off to you! 👍

kobenlys commented:
Nice Work!!! 
Thank you for adding high-priority features!! 😋😋

haerim-kweon commented on core/src/main/java/com/apighost/agent/sender/EndpointJsonBodyConverter.java:
```diff
@@ -0,0 +1,57 @@
+            if (endPoint.getRequestSchema() != null) {
+                ObjectNode root = objectMapper.createObjectNode();
+                buildJsonFromSchema(root, endPoint.getRequestSchema().getFields());
+                jsonBody = root.toString();
```
How about using objectMapper.writeValueAsString (root)?

oneweeeek replied:
I applied root.toString() → objectMapper.writeValueAsString(root)
Thank you for reminding me that I didn't consider the standard serialization method

haerim-kweon commented on core/src/main/java/com/apighost/agent/sender/EndpointJsonBodyConverter.java:
```diff
@@ -0,0 +1,57 @@
+
+    private void buildJsonFromSchema(ObjectNode node, List<FieldMeta> fields) {
+        for (FieldMeta field : fields) {
+            if (!field.getNestedFields().isEmpty()) {
```
Wouldn't it be necessary to check null?

oneweeeek replied:
Added a null check for all major objects and collections. Thank you

haerim-kweon commented on core/src/main/java/com/apighost/agent/sender/EndpointJsonBodyConverter.java:
```diff
@@ -0,0 +1,57 @@
+        return result;
+    }
+
+    private void buildJsonFromSchema(ObjectNode node, List<FieldMeta> fields) {
```
For code readability and maintenance, it would be better to divide the method by function.

oneweeeek replied:
The methods were separated by function and modified so that each method had only one responsibility.

cod0216 commented on core/src/main/java/com/apighost/agent/model/EndPoint.java:
It would be great if you could apply IntelliJ's auto-formatting!

cod0216 commented on core/src/main/java/com/apighost/agent/model/EndPointJson.java:
```diff
@@ -0,0 +1,25 @@
+package com.apighost.agent.model;
+
+public class EndPointJson {
+    private String HTTPMethod;
```
Would it be possible to use an enum here? If only limited values like POST and GET are used, I recommend using an enum.

oneweeeek committed:
refactor: improve EndpointJsonBodyConverter

Replace root.toString() with objectMapper.writeValueAsString()
Add null checks to prevent NullPointerExceptions
Enhance code readability by dividing methods by functionality
